site/_posts/2021-05-11-manageiq-perf-results.md

```diff
@@ -0,0 +1,86 @@
+![](/assets/images/blog/TotalSteadyStateCPU.png)
+
+The figure above shows the total steady state ManageIQ CPU usage across all processes/pods for three workload scenarios:
+* No providers defined
+* 1 VMware provider
+* 13 providers
+For each workload scenario, the podified CPU usage is dramatically reduced (ranging from 74 to 94% lower) with the liveness probe improvement.  With the improvement, total ManageIQ podified CPU usage is lower than appliance mode in each case.
+
+### VMDB database measurements
+The ManageIQ architecture includes the Virtual Management Database (VMDB), which is "the definitive source of intelligence collected about your Virtual Infrastructure".  During our performance measurements, we collected postgres statistics to see the total size, number of rows and activity for each of the 300+ tables in the VMDB database.
+
+The biggest tables in the vmdb database are related to metric collection.  The following observations are based on the postgres stats and discussion with development:
+* 24 raw tables, one per hour (metrics_00, metrics_01 … metrics_23)
+  * Only the most recent four hours of raw metrics are kept, and every hour, the table with data older than 4 hours is dropped.
+  * For each resource, metrics are collected every 20 seconds.  Each resource gets one metric row per collection interval, with up to 180 rows collected per hour.  
+  * For the VMware data provider, collected resources are clusters, hosts and VMs.  
+  * For the AWS data provider, the collected resources are online instances (not offline or suspended)
+  * For the Azure data provider, metrics are not collected unless the Microsoft.Insights resource provider is enabled, which was not the case in our measurements
+  * The average row size for raw metrics is 715 bytes, based on observations with the VMware data provider.
+* 12 rollup tables, one per month (metric_rollups_01, metric_rollups_02 … metric_rollups_12)
+  * Each resource gets one metric_rollups row per hour and another metric_rollup row per day.
+  * The average row size for metric_rollups is 680 bytes, based on observations with the VMware data provider.
+* vim_performance_states table
+  * This table gets one row for each insert into the metric_rollups tables.  The rows in this table represent the state of the resource when the rollup was performed. 
+  * The average row size for the vim_performance_state table is 716 bytes, base on observations with the VMware data providier.
+If metric collection is enabled, the above tables will be by far the biggest contributors to the vmdb database size.  
+To estimate the size of the metric collection portion of the database, you could use the following formula to estimate the storage in MB for one year for a given number of resources.
+ (number of resources) * ((180 raw rows/hour * 4 hours * 715 bytes/row) +  (25 rollup rows/day  * 365 days * (680 bytes/row) + (25 vim_performanced_states rows/day * 365 days * 716 bytes/row))) / 1024 /1024

```

Comment:
```suggestion
    (number of resources) * ((180 raw rows/hour * 4 hours * 715 bytes/row) +  (25 rollup rows/day  * 365 days * (680 bytes/row) + (25 vim_performanced_states rows/day * 365 days * 716 bytes/row))) / 1024 /1024`

```

Indenting like this will make it render as a block which might look nicer.